app/services/actor_services/index.rb

```diff
@@ -2,32 +2,39 @@
 
 module ActorServices
   class Index
-    attr_accessor :first_name , :last_name, :sex, :date_of_birth, :date_of_death, :movies_acted_in
+    attr_accessor :first_name , :last_name, :sex, :date_of_birth, :date_of_death, :movies_acted_in, :show_movies
 
-    def initialize(first_name: nil, last_name: nil, sex: nil, date_of_birth: nil, date_of_death: nil, movies_acted_in: nil)
+    def initialize(first_name: nil, last_name: nil, sex: nil, date_of_birth: nil, date_of_death: nil, movies_acted_in: nil, show_movies: false)
       self.first_name = first_name
       self.last_name = last_name
       self.sex = sex
       self.date_of_birth = date_of_birth
       self.date_of_death = date_of_death
       self.movies_acted_in = movies_acted_in
+      self.show_movies = show_movies
 
     end
 
     def run
+
       # get initial actors data in PDG
       actors = run_query
 
-      # build actors_ids array so it can pass into second SQL query to search for genres.
-      actor_ids = actor_ids(actors)
+      if show_movies
```

Comment:
This should be right before FillMovies and move the actor_ids into that if statement. You want to build the actor_hashes.

Also, you're getting error because you return actors which does not include 'moviesActedIn' attribute hench why you're getting error at the serializer